### ISSUE TITLE
Basic type Clock conflicts with other classes and components with the same name

### ISSUE THREAD

**Issue by OpenModelica-TracImporter** [OpenModelica/OpenModelica#3273]
Until ​https://github.com/modelica/ModelicaSpecification/issues/1538 is decided I think we should be able to handle such name clashes in the compiler by checking that the environment of the basic type Clock is top.

**Comment by AnHeuermann** [comment]
I'm not sure a model like

```modelica
model Clock
  Real x(start=1, fixed=true);
equation
  der(x)=1;
end Clock;
```

is legal Modelica code.

But in `Modelica.Electrical.Digital.Sources.Clock` from MSL 4.0.0 we have a model named `Clock` as well.

Dymola prevents me from creating a new model called `Clock` 

![image](https://user-images.githubusercontent.com/38031952/143868604-18c2d133-6084-418a-9258-32d848fb3789.png)

In OMEdit I can create the model from above but can't simulate it. Nothing will happen when I click the simulate button.

With only omc I'm getting
```bash
$ omc nameConflict.mos 
true
""

"Notification: From here:
[<interactive>:1:1-5:10:writable] Error: An element with name Clock is already declared in this scope.
"

"Notification: From here:
[<interactive>:1:1-5:10:writable] Error: An element with name Clock is already declared in this scope.
"
```

@perost @adeas31 Do we just want to return an error if someone wants to create/ load a Modelica model named `Clock` in the global scope?

**Comment by adeas31** [comment]
This is because `loadString("model Clock equation end Clock;")` is accepted by OMC.

**Comment by perost** [comment]
Not sure why you get an error about already declared elements, but the issue is that we handle `Clock` as a reserved keyword like the other builtin types. So when looking for a type `Clock` we don't even check the environment, we just check a list of reserved names and return the node for `Clock` immediately. We also do the same for the function `Clock`, but only for Modelica 3.3 and higher.

So OMC allows adding a model `Clock` to the top scope, but if you try to use it it will just find the builtin `Clock` type instead which will be instantiated as an empty model.

But I see that they've changed the specification to say that `Clock` is not reserved, so we should just put it in the top scope and look it up like any other non-reserved name. That might require a bit of work though, since `Clock` is both a type and a function (the only other such is `String` and `Integer`, which are both reserved names). I will have a look and see if I can fix it.

**Comment by perost** [comment]
Should be fixed in #8259, now we just treat it as a top-level type. So now you can do things like this:
```Modelica
package P
  record Clock
    Real t;
  end Clock;
end P;

model Clock6
  P.Clock Clock = P.Clock(1);
equation
  Clock.t = der(time);
end Clock6;
```

**Comment by perost** [comment]
Seems my commit broke Clock for the old frontend, it's using NFModelicaBuiltin in the library tests instead of ModelicaBuiltin like it should. That's bad and I don't know why it happens, but I made a workaround in #8261 until I have time to fix it properly. It shouldn't really affect anything except change the error message a bit when trying to declare a top-level Clock element.
